mediapipe/tasks/python/vision/image_classifier.py

```diff
@@ -0,0 +1,218 @@
+        task_graph=_TASK_GRAPH_NAME,
+        input_streams=[
+            ':'.join([_IMAGE_TAG, _IMAGE_IN_STREAM_NAME]),
+        ],
+        output_streams=[
+            ':'.join([_CLASSIFICATION_RESULT_TAG,
+                      _CLASSIFICATION_RESULT_OUT_STREAM_NAME]),
+            ':'.join([_IMAGE_TAG, _IMAGE_OUT_STREAM_NAME])
+        ],
+        task_options=options)
+    return cls(
+        task_info.generate_graph_config(
+            enable_flow_limiting=options.running_mode ==
+            _RunningMode.LIVE_STREAM), options.running_mode,
+        packets_callback if options.result_callback else None)
+
+  # TODO: Create an Image class for MediaPipe Tasks.
+  def classify(
+      self,
+      image: image_module.Image,
+  ) -> classifications_module.ClassificationResult:
+    """Performs image classification on the provided MediaPipe Image.
+
+    Args:
+      image: MediaPipe Image.
+
+    Returns:
+      A classification result object that contains a list of classifications.
+
+    Raises:
+      ValueError: If any of the input arguments is invalid.
+      RuntimeError: If image classification failed to run.
+    """
+    output_packets = self._process_image_data(
+        {_IMAGE_IN_STREAM_NAME: packet_creator.create_image(image)})
+    classification_result_proto = packet_getter.get_proto(
+        output_packets[_CLASSIFICATION_RESULT_OUT_STREAM_NAME])
+
+    return classifications_module.ClassificationResult([
+        classifications_module.Classifications.create_from_pb2(classification)
+        for classification in classification_result_proto.classifications
+    ])
+
+  def classify_for_video(
+      self, image: image_module.Image,
+      timestamp_ms: int
+  ) -> classifications_module.ClassificationResult:
+    """Performs image classification on the provided video frames.
+
+    Only use this method when the ImageClassifier is created with the video
+    running mode. It's required to provide the video frame's timestamp (in
+    milliseconds) along with the video frame. The input timestamps should be
+    monotonically increasing for adjacent calls of this method.
+
+    Args:
+      image: MediaPipe Image.
+      timestamp_ms: The timestamp of the input video frame in milliseconds.
+
+    Returns:
+      A classification result object that contains a list of classifications.
+
+    Raises:
+      ValueError: If any of the input arguments is invalid.
+      RuntimeError: If image classification failed to run.
+    """
+    output_packets = self._process_video_data({
+      _IMAGE_IN_STREAM_NAME:
+        packet_creator.create_image(image).at(timestamp_ms)
+    })
+    classification_result_proto = packet_getter.get_proto(
+      output_packets[_CLASSIFICATION_RESULT_OUT_STREAM_NAME])
+
+    return classifications_module.ClassificationResult([
+        classifications_module.Classifications.create_from_pb2(classification)
+        for classification in classification_result_proto.classifications
+    ])
```

Comment:
If possible, please add the classify_async api into this PR and update the unit tests.

Reply:
@jiuqiant I encountered the following while trying to run tests for the `classify_async` API.

```
F20221008 05:19:47.966590    64 packet.cc:138] Check failed: holder_ != nullptr The packet is empty.
```

I'll include it in this PR with code the code commented out so you try reproducing the issue. Everything else seems to run fine.

Reply:
MediaPipe returns empty packets if there is nothing detected. Probably we need to check if the packet is empty, do something like:

```
if output_packets[_CLASSIFICATION_RESULT_OUT_STREAM_NAME].is_empty():
  # return empty classification result
else:
  # make a classification result from packet.
```

Reply:
@jiuqiant Thanks. I was missing this check and now the API runs fine after adding it.